code/cbr.py

```diff
@@ -488,9 +511,13 @@ def main(args):
     parser = argparse.ArgumentParser(description="Run CBR")
     parser.add_argument("--dataset_name", type=str, help="The dataset name. Replace with one of FB122 | WN18RR | NELL-995 to reproduce the results of the paper")
     parser.add_argument("--data_dir", type=str, default="./cbr-akbc-data/")
+    parser.add_argument("--output_dir", type=str, default="./output/")
     parser.add_argument("--test", action="store_true")
+    parser.add_argument("--parallelize", action="store_true")
     parser.add_argument("--test_file_name", type=str, default='')
     parser.add_argument("--max_num_programs", type=int, default=15, help="Max number of paths to consider")
+    parser.add_argument("--splitid", type=int, default=0, help="Split number")
```

Comment:
Good job overall. But I think design is a bit flawed. Instead just take argument as the total number of workers  and the split id.